src/plugins/intel_gpu/src/plugin/ops/mlir_op.cpp

```diff
@@ -38,21 +51,71 @@ void CreateMLIRSubgraphOp(ProgramBuilder& p, const std::shared_ptr<ov::op::mlir:
 
         ov::TensorVector input_host_tensors;
         ov::TensorVector output_host_tensors;
+        // to keep cl_mem objects alive
+        std::vector<cl_mem> cl_mem_refs;
+
+        auto process_buffer = [&stream, &cl_mem_refs](cldnn::memory::ptr mem, ov::TensorVector& tensors) {
+            switch (mem->get_allocation_type()) {
+                case cldnn::allocation_type::cl_mem: {
+                    auto gpu_buff = dynamic_cast<cldnn::ocl::gpu_buffer*>(mem.get());
+                    cl_mem cl_buff = gpu_buff->get_buffer().get();
+
+                    // Keep the cl_mem reference alive
+                    cl_mem_refs.push_back(cl_buff);
+                    tensors.push_back(make_tensor(mem->get_layout(), &cl_mem_refs.back()));
+                    std::cout << " cl_mem: " << &cl_buff;
+                    break;
+                }
+                case cldnn::allocation_type::usm_host:
+                case cldnn::allocation_type::usm_shared:
+                case cldnn::allocation_type::usm_device: {
+                    auto usm_ptr = mem->buffer_ptr();
+                    auto gpu_buff = dynamic_cast<cldnn::ocl::gpu_usm*>(mem.get());
+                    auto& usm_helper = gpu_buff->get_buffer().getUsmHelper();
+                    // HACK: force move to device, can we do better than this?
+                    usm_helper.enqueue_memcpy(
+                        dynamic_cast<cldnn::ocl::ocl_stream&>(stream).get_cl_queue(),
+                        usm_ptr,
+                        usm_ptr,
+                        mem->get_layout().bytes_count());
+                    std::cout << " usm_ptr: " << usm_ptr;
+                    tensors.push_back(make_tensor(mem->get_layout(), usm_ptr));
```

Comment:
Could you try to enforce `stream.finish();` for any queue type instead? Maybe incorrect events list is passed to this op.

Reply:
Well, now I can't reproduce the problem because the queue is now appears to be sequential only. Is there a way to enforce it to be out-of-order in OV?